maven-extension/src/main/java/io/opentelemetry/maven/OtelExecutionListener.java

```diff
@@ -200,8 +245,14 @@ public void mojoStarted(ExecutionEvent executionEvent) {
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_ID, mojoExecution.getExecutionId())
             .setAttribute(
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_LIFECYCLE_PHASE,
-                mojoExecution.getLifecyclePhase())
-            .startSpan();
+                mojoExecution.getLifecyclePhase());
+    //  enrich spans with MojoGoalExecutionHandler
+    Optional<MojoGoalExecutionHandler> handler =
```

Comment:
We don't use Optional much in OTel codebases in favor of Nullable. I just noticed I've been forgetting to add nullness checking to this repo like we have in SDK repo, will add that. Can you look into reducing Optional for more consistency with SDK repo? I don't think the usages here are all that different from null checks

Reply:
I like the principle _"When in Rome, do as the Romans do"_ :-). I'll replace the usage of `Optional` by using `@Nullable`

Reply:
Migration from `Optional` to `@Nullable` done. Please verify